src/web/core/src/main/java/org/geoserver/web/data/store/panel/FileModel.java

```diff
@@ -50,8 +51,14 @@ private boolean isSubfile(File root, File selection) {
         return isSubfile(root, selection.getParentFile());
     }
 
+    @Override
     public String getObject() {
-        return delegate.getObject();
+        Object obj = delegate.getObject();
```

Comment:
Indeed the "full" fix would be to change the REST code to insert a string instead of a URL. However I wasn't confident enough to find all the places where it potentially relied on the existing behaviour.